src/home/pam_systemd_home.c

```diff
@@ -625,6 +625,15 @@ static int acquire_home(
                 do_auth = true;
         }
 
+        /* Later PAM modules may need the auth token, but only during pam_authenticate. */
+        if (please_authenticate && secret->password && *secret->password) {
```

Comment:
Not sure, but maybe `secret->password && *secret->password` -> `!strv_isempty(secret->password)`?

Reply:
Ok.